viadot/sources/cloud_for_customers.py

```diff
@@ -0,0 +1,257 @@
+import re
+import requests
+import pandas as pd
+
+from copy import deepcopy
+from typing import Any, Dict, List, Optional
+from pydantic import BaseModel
+from urllib.parse import urljoin
+from viadot.exceptions import CredentialError
+
+from ..utils import handle_api_response
+from ..config import get_source_credentials
+from .base import Source
+
+
+class CloudForCustomersCredentials(BaseModel):
+    username: str  # eg. username@{tenant_name}.com
+    password: str
+
+
+class CloudForCustomers(Source):
+    """Cloud for Customers connector to fetch Odata source.
+
+    Args:
+        url (str, optional): The URL to the C4C API. E.g 'https://my396876543.jrt.ondemand.com/c4c/v1/'.
+        endpoint (str, optional): The API endpoint.
+        report_url (str, optional): The API URL in case of prepared report.
+        filter_params (Dict[str, Any], optional): Filtering parameters passed to the request. E.g {"$filter": "AccountID eq '1234'"}.
+        More info on: https://userapps.support.sap.com/sap/support/knowledge/en/2330688
```

Comment:
tabulate this line

Reply:
This is what I mean when I say assertive communication. It is not obvoius where do you want the tabulation and on which line of code. it is intuitable but not unambiguous. I would have to push a wrong tabulation or ask you and then wait for your answer to know exactly what change you want. Maybe im dumb and I can not recognize the obvious tabulation that is needed here (it is a possibility) but it does not matter, as a rule of thumb, communication should be foolproof

This could be avoided if the comment would have been something like: "tabulate line 29, between x and y" or something like that.

Hope you take this as constructive feedback because it was given with that intention.🙂

Reply:
I will make the tabulation between : and https